src/Widget/Canvas/Grid/BrailleGrid.php

```diff
@@ -66,4 +75,13 @@ public function paint(Position $position, Color $color): void
             $this->colors[$index] = $color;
         }
     }
+
+    private function cacheBrailleChars(): void
+    {
+        [$from, $to] = BrailleSet::RANGE;
+
+        for ($i = $from; $i <= $to; $i++) {
+            $this->brailleCharCache[$i] = IntlChar::chr($i);
```

Comment:
hm, is there any point in doing this and lazily loading the char on line 54? i think we could remove this function no?

Reply:
Yes, I think this is a very good idea. Should I send a PR?

Reply:
it's fine,I can do it quickly I think

Reply:
done

```diff
--- a/src/Model/Widget/BrailleSet.php
+++ b/src/Model/Widget/BrailleSet.php
@@ -4,8 +4,6 @@ namespace PhpTui\Tui\Model\Widget;

 final class BrailleSet
 {
-    // Braille patterns range from U+2800 to U+28FF
-    public const RANGE = [0x2800, 0x28FF];
     public const BLANK = 0x2800;
     public const DOTS = [
         [0x0001, 0x0008],
diff --git a/src/Widget/Canvas/Grid/BrailleGrid.php b/src/Widget/Canvas/Grid/BrailleGrid.php
index 6cfd70b..eb3f1fc 100644
--- a/src/Widget/Canvas/Grid/BrailleGrid.php
+++ b/src/Widget/Canvas/Grid/BrailleGrid.php
@@ -29,7 +29,6 @@ final class BrailleGrid extends CanvasGrid
         private array $codePoints,
         private array $colors
     ) {
-        $this->cacheBrailleChars();
     }

     public static function new(int $width, int $height): self
@@ -75,13 +74,4 @@ final class BrailleGrid extends CanvasGrid
             $this->colors[$index] = $color;
         }
     }
-
-    private function cacheBrailleChars(): void
-    {
-        [$from, $to] = BrailleSet::RANGE;
-
-        for ($i = $from; $i <= $to; $i++) {
-            $this->brailleCharCache[$i] = IntlChar::chr($i);
-        }
-    }
 }
```

Reply:
Awesome! I've send a PR with a :nail_care: change: https://github.com/php-tui/php-tui/pull/67